x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/test_pipeline.tsx

```diff
@@ -7,6 +7,110 @@
 
 import React from 'react';
 
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiCodeBlock,
+  EuiResizableContainer,
+  EuiButton,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  useIsWithinMaxBreakpoint,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { CodeEditor } from '@kbn/kibana-react-plugin/public';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+
 export const TestPipeline: React.FC = () => {
-  return <div>Test Pipeline</div>;
+  const {
+    addInferencePipelineModal: { simulateBody },
+    simulatePipelineResult,
+    simulatePipelineErrors,
+  } = useValues(MLInferenceLogic);
+  const { simulatePipeline, setPipelineSimulateBody } = useActions(MLInferenceLogic);
+
+  const isSmallerViewport = useIsWithinMaxBreakpoint('s');
+
+  return (
+    <EuiFlexGroup direction="column" gutterSize="xs">
+      <EuiFlexItem>
+        <EuiText>
+          <h4>
+            {i18n.translate(
+              'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.test.title',
+              { defaultMessage: 'Review pipeline results (optional)' }
+            )}
+          </h4>
+        </EuiText>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiResizableContainer
+          direction={isSmallerViewport ? 'vertical' : 'horizontal'}
+          style={{ height: 310 }}
```

Comment:
question: all this explicit size styling gives me the heebie-jeebies, is there a better (more responsive) way to achieve the same goal?